hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -309,9 +314,23 @@ def _define_setting(
         if hide_repr is not_set:
             hide_repr = bool(deprecation_message)
 
+        if deprecation_message is not None and deprecated_since is None:
+            note_deprecation(
+                'Settings with a deprecation message should always include '
+                'a `deprecated_since` date.',
+                since='2018-12-17'
```

Comment:
This can safely be `"2018-10-27"`, as we haven't merged any deprecations since then.